[ONNX] Introduce ONNXScopeName

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82040
* #82039
* __->__ #82038

Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

BowenBao committed:
[ONNX] Introduce ONNXScopeName

[ghstack-poisoned]

justinchuby commented on torch/onnx/utils.py:
```diff
@@ -1251,12 +1251,10 @@ def unconvertible_ops(
+def _setup_trace_module_map(
+    model: Union[torch.nn.Module, torch.jit.ScriptModule],
+    export_modules_as_functions: Union[bool, Collection[Type[torch.nn.Module]]],
+):
```
```suggestion
) -> bool:
```
does it return bool?

AllenTiTaiWang replied:
I think it's not. From the if and elif branch, it seems like could be dict.

BowenBao replied:
updated to `Set[str]`

justinchuby commented on torch/onnx/utils.py:
```diff
@@ -1280,9 +1278,15 @@ def _track_module_attributes_forward_hook(module, input, output):
-        trace_module_map = __setup_trace_module_map()
-        export_modules_as_functions = {v for k, v in trace_module_map.items()}
+        export_modules_as_functions = {
+            torch.typename(type(_m)) for _m, _ in trace_module_map.items()
```
```suggestion
            torch.typename(type(named_module)) for named_module in trace_module_map
```

maybe this for simplicity?

justinchuby commented on torch/onnx/utils.py:
```diff
@@ -1280,9 +1278,15 @@ def _track_module_attributes_forward_hook(module, input, output):
+        }
     elif (
         isinstance(export_modules_as_functions, set)
         and len(export_modules_as_functions) > 0
```
```suggestion
        and export_modules_as_functions
```
nit: unrelated to this pr but

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"




[ghstack-poisoned]

AllenTiTaiWang commented on torch/onnx/utils.py:
```diff
@@ -1298,11 +1302,10 @@ def _find_typename(v):
         export_modules_as_functions = module_typenames
     else:
-        export_modules_as_functions = None
+        export_modules_as_functions = False
```
I wonder if there is a reason `export_modules_as_functions` should be a bool if it's not a dict? I see it's later consumed by `_jit_pass_onnx_function_extraction`

BowenBao replied:
I think it was a mistake. Now it's renamed, and assigned as empty set.

titaiwangms reviewed (approved):
:beers:

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

justinchuby commented on torch/csrc/jit/passes/onnx/naming.cpp:
```diff
@@ -0,0 +1,80 @@
+
+namespace {
+
+std::string nameFromRoot(
```
Should functions follow PascalCase naming?

justinchuby replied:
I realized PyTorch uses camel case for functions everywhere else. Sorry for making the previous suggestion

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao commented:
cc @abock for merge approval

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge -g

BowenBao commented:
@malfet could you help merge this one? I think it's touching `build_variables.bzl`

BowenBao commented:
@pytorchbot merge -g

BowenBao committed:
Update on "[ONNX] Introduce ONNXScopeName"


Update `_setup_trace_module_map` to always record module/layer info
in `Scope` attribute for nodes.
Extend `Scope` name to not only record module typename, but also
module object variable name. Both names are formatted and stored
as `name` attribute in `Scope`.
Introduce `ONNXScopeName` class to manage the formatting and parsing.
Updated local function export code adjusting to this update.

[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge -g
